keys/keys.go

```diff
@@ -261,10 +261,10 @@ func ValidateRangeMetaKey(key proto.Key) error {
 	prefix, body := proto.Key(key[:len(Meta1Prefix)]), proto.Key(key[len(Meta1Prefix):])
 
 	if prefix.Equal(Meta2Prefix) {
-		if body.Less(proto.KeyMax) {
-			return nil
+		if proto.KeyMax.Less(body) {
+			return NewInvalidRangeMetaKeyError("body of meta2 range lookup is > KeyMax", key)
```

Comment:
sorry, didn't read this right the first time. This is actually comparing for `>` (note that the old code returned `nil` if the condition matches)